src/arrayMethodJoin.js

```diff
@@ -4,9 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        result += separator;
+      } else {
+        result += this[i];
+        result += separator;
+      }
+    }
```

Comment:
```suggestion
    for (let i = 0; i < this.length; i++) {
      if (this[i] !== null || this[i] !== undefined) {
        result += this[i];
      }
      
      if (think what condition to use) {
        result += separator;
      }
    }
```

This way you could delete lines 20-26.